libp2p/varint.nim

```diff
@@ -158,9 +207,40 @@ proc putUVarint*[T: PB|LP](vtype: typedesc[T],
 
 proc getSVarint*(pbytes: openarray[byte], outsize: var int,
                  outval: var PBSomeSVarint): VarintStatus {.inline.} =
-  ## Decode Google ProtoBuf's `signed varint` from buffer ``pbytes`` and store
-  ## it to ``outval``. On success ``outlen`` will be set to number of bytes
-  ## processed while decoding `signed varint`.
+  ## Decode signed integer (``int32`` or ``int64``) from buffer ``pbytes``
+  ## and store it to ``outval``.
+  ##
+  ## On success ``outlen`` will be set to number of bytes processed while
+  ## decoding signed varint.
+  ##
+  ## If array ``pbytes`` is empty, ``Incomplete`` error will be returned.
+  ##
+  ## If there not enough bytes available in array ``pbytes`` to decode `signed
+  ## varint`, ``Incomplete`` error will be returned.
+  ##
+  ## If encoded value can produce integer overflow, ``Overflow`` error will be
+  ## returned.
+  ##
+  ## Note, when decoding 10th byte of 64bit integer only 1 bit from byte will be
+  ## decoded, all other bits will be ignored. When decoding 5th byte of 32bit
+  ## integer only 4 bits from byte will be decoded, all other bits will be
+  ## ignored.
+  when sizeof(outval) == 8:
+    var value: uint64
+  else:
+    var value: uint32
+
+  result = PB.getUVarint(pbytes, outsize, value)
+  if result == VarintStatus.Success:
+    outval = cast[type(outval)](value)
+
+proc getSVarint*(pbytes: openarray[byte], outsize: var int,
```

Comment:
since there's a dedicated type for each of unsigned, cast and zigzag, could have a single `getVarint` now

Reply:
this would make the interface consistent across vsizeof and get/put etc